Gems/SceneProcessing/Code/Tests/MeshBuilder/MeshOptimizerComponentTests.cpp

```diff
@@ -0,0 +1,221 @@
+
+                // This assumes that the data coming from the import process gives a unique control point
+                // index to every vertex. This follows the behavior of the AssImp library.
+                mesh->SetVertexIndexToControlPointIndexMap(i, i);
+                ++i;
+            }
+
+            mesh->AddFace({0, 1, 2}, 0);
+            mesh->AddFace({3, 4, 5}, 0);
+
+            return mesh;
+        }
+
+        static AZStd::unique_ptr<AZ::SceneData::GraphData::SkinWeightData> MakeSkinData()
+        {
+            auto skinWeights = AZStd::make_unique<AZ::SceneData::GraphData::SkinWeightData>();
+
+            skinWeights->ResizeContainerSpace(6);
+
+            // Add bones 0 and 1 to the skin weights
+            skinWeights->GetBoneId("0");
+            skinWeights->GetBoneId("1");
+
+            skinWeights->AppendLink(0, {/*.boneId=*/0, /*.weight=*/1});
+            skinWeights->AppendLink(1, {/*.boneId=*/0, /*.weight=*/1});
+            skinWeights->AppendLink(2, {/*.boneId=*/0, /*.weight=*/1});
+            skinWeights->AppendLink(3, {/*.boneId=*/1, /*.weight=*/1});
+            skinWeights->AppendLink(4, {/*.boneId=*/1, /*.weight=*/1});
+            skinWeights->AppendLink(5, {/*.boneId=*/1, /*.weight=*/1});
+
+            return skinWeights;
+        }
+
+    private:
+        AZ::ComponentApplication m_app;
+        AZ::Entity* m_systemEntity;
+    };
+
+    TEST_F(VertexDeduplicationFixture, CanDeduplicateVertices)
+    {
+        AZ::SceneAPI::Containers::Scene scene("testScene");
+        AZ::SceneAPI::Containers::SceneGraph& graph = scene.GetGraph();
+
+        const auto meshNodeIndex = graph.AddChild(graph.GetRoot(), "testMesh", MakePlaneMesh());
+
+        // The original source mesh should have 6 vertices
+        EXPECT_EQ(AZStd::rtti_pointer_cast<AZ::SceneAPI::DataTypes::IMeshData>(graph.GetNodeContent(meshNodeIndex))->GetVertexCount(), 6);
+
+        auto meshGroup = AZStd::make_unique<AZ::SceneAPI::SceneData::MeshGroup>();
+        meshGroup->GetSceneNodeSelectionList().AddSelectedNode("testMesh");
+        scene.GetManifest().AddEntry(AZStd::move(meshGroup));
+
+        AZ::SceneGenerationComponents::MeshOptimizerComponent component;
+        AZ::SceneAPI::Events::GenerateSimplificationEventContext context(scene, "pc");
+        component.OptimizeMeshes(context);
+
+        AZ::SceneAPI::Containers::SceneGraph::NodeIndex optimizedNodeIndex = graph.Find(AZStd::string("testMesh").append(AZ::SceneAPI::Utilities::OptimizedMeshSuffix));
+        ASSERT_TRUE(optimizedNodeIndex.IsValid()) << "Mesh optimizer did not add an optimized version of the mesh";
+
+        const auto& optimizedMesh = AZStd::rtti_pointer_cast<AZ::SceneAPI::DataTypes::IMeshData>(graph.GetNodeContent(optimizedNodeIndex));
+        ASSERT_TRUE(optimizedMesh);
+
+        // The optimized mesh should have 4 vertices, the 2 shared vertices are welded together
+        EXPECT_EQ(optimizedMesh->GetVertexCount(), 4);
```

Comment:
It would be helpful to verify the 4 vertices match expected values. If something breaks internally where it welds the wrong vertices and still outputs 4, this test would pass as-is.